### PR TITLE
Add documentation on how to configure proxies

### PR DESCRIPTION
**Pull request by eqrx**
**What this PR does / why we need it**:
Improve documentation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
None

**Special notes for your reviewer**:
None
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/proxy.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# How to use KubeOne in a proxy environment
+
+Create a KubeOne configuration as usual and add the following section:
+
+```yaml
+proxy:
+  http_proxy: 'http://proxy.example.com'
```
Comment (chrkl) [inline docs/proxy.md]: This doesn't match the `kubeone config print` output:
proxy:
 http: ''
 https: ''
 noProxy: ''

Comment (kron4eg) [reply docs/proxy.md]: indeed, this is pre-0.6 config format. it's time to forget about it once and for all.

**Comment by toschneck** [comment]
@eqrx can you follow up on this?

**Comment by xmudrii** [comment]
Agreed with @eqrx to close this PR since it's old and out-of-sync.

**Comment by toschneck** [comment]
@xmudrii did we finally documented it somewhere. This could cost a lot of time at customer installation, so we really really should have a documented working example.
The best would also to have an e2e test for the proxy support.

**Comment by xmudrii** [comment]
@toschneck No, we don't have any proxy-related document currently, but we would like to. I think it would be best that someone who has tested and used the proxy feature create a PR to document it.

**Comment by xmudrii** [comment]
@toschneck I've created a follow-up issue #739.

**Pull request closed without merging**